core/kernel/interrupt.c

```diff
@@ -37,51 +41,205 @@
  */
 
 static struct itr_chip *itr_chip;
-static SLIST_HEAD(, itr_handler) handlers = SLIST_HEAD_INITIALIZER(handlers);
+static struct itr_desc *itr_descs[256] = {0};
+static unsigned itr_lock;
```

Comment:
Do we need dynamic registration/unregistration of interrupt handlers? Wouldn't it be enough to be able to enable/disable them dynamically?